### PR TITLE
Smoother animation towards the end of opening/closing pinny

### PR DESCRIPTION
**Pull request by vmarta**
Status: open for review
Reviewer: @Helen-Mobify 

The opening/closing animation of pinny is slowed down by the things we do in `openComplete()` and `closeComplete()`. See a screenshot of this slowdown below.

So I added a setTimeout of 0 to delay them and to avoid conflicting with the animation. Tested on iPad 3 running iOS 8.0.

Timeline tab

### CODE REVIEW

**Comment by vmarta** [comment]
This is what the timeline looks like _after_ the fix:

**Comment by Helen-Mobify** [comment]
@vmarta This looks amazing

Pulling a couple people in to look at this together. @fractaltheory @donnielrt

**Comment by donnielrt** [comment]
Very cool, @vmarta!

**Comment by donnielrt** [comment]
Shouldn't openComplete be firing *after* the animation in any case?

File: dist/pinny.js [hunk 1]
```diff
@@ -100,49 +100,54 @@
          */
         animation: {
             openComplete: function() {
-                var plugin = this;
-                this._disableExternalInputs();
-                this._focus();
+                setTimeout(function () {
```
Comment (donnielrt) [inline dist/pinny.js]: We could also test with requestAnimationFrame here instead of setTimeout

**Comment by mlworks** [comment]
@donnielrt Yea it is weird how Velocity's Complete callback happens BEFORE the animation finishes. Here's a screenshot with a `debugger` set in the complete callback. Notice the small gap in the left, animation is still not done.

screen shot 2015-06-25 at 3 14 17 pm

**Comment by donnielrt** [comment]
@mlworks interesting, that makes me think we're maybe dealing with a bug we hadn't caught before. If openComplete is firing **before** the open animation is complete, that's a major bug!

**Comment by donnielrt** [comment]
Hey, we’re looking to prune older abandoned PRs. If this PR is still relevant and you would like to see it merged in, please reopen the PR and we’ll add it to our backlog! Thanks!

**Comment by vmarta** [comment]
Let's reopen this baby! I feel we got a solid fix to pinny's animation.. we've been using it for a couple of our recent projects. 

What do we need to do next to get this PR approved?

**Comment by donnielrt** [comment]
Thanks @vmarta! We're going to take a quick look at it, and get in touch if we have questions. Otherwise we'll go ahead and prep this for publishing.

File: dist/pinny.js (hunk 1 above)
Comment (haroldtreen) [inline dist/pinny.js]: Ya, this example makes it look like RAF is more efficient http://jsfiddle.net/calpo/h7eee/.

Comment (haroldtreen) [inline dist/pinny.js]: @donnielrt 

Nvm. Using RAF causes the mocha tests to fail :(. This seems good enough :+1:

**Comment by haroldtreen** [comment]
Thanks for the PR @vmarta. 

LGTM :+1:

**Comment by haroldtreen** [comment]
Merged this PR into `btr-fixes` where a bunch of other changes are waiting to do a release. We'll get this in through that PR @vmarta  :).

https://github.com/mobify/pinny/commit/412fc34a230a54c1b01cc01267ddd0718870e246

**Comment by vmarta** [comment]
@haroldtreen sounds good.. thanks Harold 👍🏻

**Pull request closed without merging**